src/types/onyx/Report.ts

```diff
@@ -178,6 +179,8 @@ type Report = {
     reportFields?: Record<string, PolicyReportField>;
 };
 
+type ReportCollectionDataSet = Record<`${typeof ONYXKEYS.COLLECTION.REPORT}${string}`, Report>;
```

Comment:
WDYT about creating a generic `CollectionDataSet` type under `src/types/utils` with a comment explaining that can be useful for `multiSet` calls or something like that?

```ts
type CollectionDataSet<TCollectionKey extends OnyxCollectionKey> = Record<`${TCollectionKey}${string}`, OnyxCollectionValuesMapping[TCollectionKey]>;
```

And then you can use it here:

```ts
type ReportCollectionDataSet = CollectionDataSet<typeof ONYXKEYS.COLLECTION.REPORT>;
```